src/OpenApi/src/Extensions/JsonObjectSchemaExtensions.cs

```diff
@@ -0,0 +1,279 @@
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Constraints;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+/// <summary>
+/// Provides a set of extension methods for modifying the opaque JSON Schema type
+/// that is provided by the underlying schema generator in System.Text.Json.
+/// </summary>
+internal static class JsonObjectSchemaExtensions
+{
+    private static readonly Dictionary<Type, OpenApiSchema> _simpleTypeToOpenApiSchema = new()
+    {
+        [typeof(bool)] = new() { Type = "boolean" },
+        [typeof(byte)] = new() { Type = "string", Format = "byte" },
+        [typeof(int)] = new() { Type = "integer", Format = "int32" },
+        [typeof(uint)] = new() { Type = "integer", Format = "int32" },
+        [typeof(long)] = new() { Type = "integer", Format = "int64" },
+        [typeof(ulong)] = new() { Type = "integer", Format = "int64" },
+        [typeof(short)] = new() { Type = "integer", Format = null },
+        [typeof(ushort)] = new() { Type = "integer", Format = null },
+        [typeof(float)] = new() { Type = "number", Format = "float" },
+        [typeof(double)] = new() { Type = "number", Format = "double" },
+        [typeof(decimal)] = new() { Type = "number", Format = "double" },
+        [typeof(DateTime)] = new() { Type = "string", Format = "date-time" },
+        [typeof(DateTimeOffset)] = new() { Type = "string", Format = "date-time" },
+        [typeof(Guid)] = new() { Type = "string", Format = "uuid" },
+        [typeof(char)] = new() { Type = "string" },
+
+        // Nullable types
+        [typeof(bool?)] = new() { Type = "boolean", Nullable = true },
+        [typeof(byte?)] = new() { Type = "string", Format = "byte", Nullable = true },
+        [typeof(int?)] = new() { Type = "integer", Format = "int32", Nullable = true },
+        [typeof(uint?)] = new() { Type = "integer", Format = "int32", Nullable = true },
+        [typeof(long?)] = new() { Type = "integer", Format = "int64", Nullable = true },
+        [typeof(ulong?)] = new() { Type = "integer", Format = "int64", Nullable = true },
+        [typeof(short?)] = new() { Type = "integer", Format = null, Nullable = true },
+        [typeof(ushort?)] = new() { Type = "integer", Format = null, Nullable = true },
+        [typeof(float?)] = new() { Type = "number", Format = "float", Nullable = true },
+        [typeof(double?)] = new() { Type = "number", Format = "double", Nullable = true },
+        [typeof(decimal?)] = new() { Type = "number", Format = "double", Nullable = true },
+        [typeof(DateTime?)] = new() { Type = "string", Format = "date-time", Nullable = true },
+        [typeof(DateTimeOffset?)] = new() { Type = "string", Format = "date-time", Nullable = true },
+        [typeof(Guid?)] = new() { Type = "string", Format = "uuid", Nullable = true },
+        [typeof(char?)] = new() { Type = "string", Nullable = true },
+        [typeof(Uri)] = new() { Type = "string", Format = "uri" },
```

Comment:
Do these not need `Nullable`?  Maybe `string` is nullable in the destination representation?

Reply:
`Uri` and `string` are both nullable reference types so there's no underlying runtime type here. But, this question made me realize that we don't need to set the nullable property here at all. The handling that we do in `OpenApiJsonSchema` extensions is sufficient because the underlying schema generator does the work to read the NullabilityContext on the types.